tests/test_query.py

```diff
@@ -561,6 +563,34 @@ def test_get_invalid_pipeline_name(
     assert response.status_code == 422
 
 
+@pytest.mark.parametrize(
+    "valid_pipeline_name",
+    ["np:fmriprep", "np:fmriprep", "np:freesurfer", "np:freesurfer"],
+)
+@pytest.mark.parametrize(
+    "valid_pipeline_version",
+    ["v2.0.1", "23.1.3", "7.3.2", "8.7.0-rc"],
+)
```

Comment:
🍒: I think the following would be more efficient than having two parametrize statements, which results in **every** combination of the two lists being tested. (Since you have duplicates in the pipeline names, you are currently running the same combinations twice of fmriprep and freesurfer version pairings twice.)

```suggestion
@pytest.mark.parametrize(
    "valid_pipeline_name, valid_pipeline_version",
    [
        ("np:fmriprep", "v2.0.1"),
        ("np:fmriprep", "23.1.3"),
        ("np:freesurfer", "7.3.2"),
        ("np:freesurfer", "8.7.0-rc"),
    ],
)
```
I think these example pairings also sufficiently cover different possible version formats :)